tests/core/data_layer/test_data_store.py

```diff
@@ -1433,3 +1433,137 @@ async def test_benchmark_batch_insert_speed(
             tree_id=tree_id,
             changelist=batch,
         )
+
+
+@pytest.mark.anyio
+async def test_delete_store_data(raw_data_store: DataStore) -> None:
+    tree_id = bytes32(b"\0" * 32)
+    tree_id_2 = bytes32(b"\0" * 31 + b"\1")
+    await raw_data_store.create_tree(tree_id=tree_id, status=Status.COMMITTED)
+    await raw_data_store.create_tree(tree_id=tree_id_2, status=Status.COMMITTED)
+    total_keys = 4
+    keys = [key.to_bytes(4, byteorder="big") for key in range(total_keys)]
+    batch1 = [
+        {"action": "insert", "key": keys[0], "value": keys[0]},
+        {"action": "insert", "key": keys[1], "value": keys[1]},
+    ]
+    batch2 = batch1.copy()
+    batch1.append({"action": "insert", "key": keys[2], "value": keys[2]})
+    batch2.append({"action": "insert", "key": keys[3], "value": keys[3]})
+    await raw_data_store.insert_batch(tree_id, batch1, status=Status.COMMITTED)
+    await raw_data_store.insert_batch(tree_id_2, batch2, status=Status.COMMITTED)
+    assert batch1 != batch2
```

Comment:
Certainly a nitpick.  Slightly more clear though I think to have the assertion immediately adjacent to the variation.

```suggestion
    assert batch1 != batch2
    await raw_data_store.insert_batch(tree_id, batch1, status=Status.COMMITTED)
    await raw_data_store.insert_batch(tree_id_2, batch2, status=Status.COMMITTED)
```